lightkurve/correctors/designmatrix.py

```diff
@@ -264,16 +286,44 @@ def values(self):
         return self.df.values
 
     def __getitem__(self, key):
-        return self.df[key]
+        return self.df[key].values
 
     def __repr__(self):
         return '{} DesignMatrix {}'.format(self.name, self.shape)
 
+    def to_sparse(self):
+        return SparseDesignMatrix(csr_matrix(self.values), name=self.name,
+                            columns=self.columns, prior_mu=self.prior_mu,
+                             prior_sigma=self.prior_sigma)
+
+    def join(self, matrix):
+        """ Join two designmatrices, return a design matrix collection """
+        return self.__class__([self, matrix])
+
+    def multichoose(self):
+        """ Create products of each vector in the design matrix """
+        raise NotImplementedError
+
 
 class DesignMatrixCollection():
     """A set of design matrices."""
-    def __init__(self, matrices):
-        self.matrices = matrices
+    def __init__(self, matrices, warn=True):
```

Comment:
The `warn` parameter should be removed.  We should stick to one of two options for controlling warnings:
* If a user is making a mistake and should fix their code, use `warnings.warn("message", LightkurveWarning)` and let them silence it using `catch_warnings()` if they want to.  As a general rule we should only emit warnings if we think the user HAS to fix their code (e.g. deprecation warning).  We shouldn't provide convenient options to silence these.
* To record messages to help diagnose what Lightkurve is doing and to help the user, use `log.debug()` or `log.info()`.  The user can turn these on or off using `lk.log.setLevel()`.
(Details: 
https://github.com/KeplerGO/lightkurve/issues/691#issuecomment-586016365)

I'd use `log.debug()`, unless you think this is really stupid, in which case always emit a warning.

Reply:
I'm leaving it as a warning. Mixing sparse and non-sparse matrices is always dangerous, but there are reasons that a user may want to do it. Thanks for the explanation!